soc/xtensa/intel_adsp/common/adsp.c

```diff
@@ -149,7 +43,6 @@ static const struct adsp_ipc_fw_ready fw_ready_apl
 	.flags = 0,
 };
 
-#if !defined(CONFIG_SOF)
```

Comment:
this `#if` is unrelated to GPIOs and is needed for SOF, removing it breaks audio on UP2. Please, remove this and the following hunks from this patch